lib/logstash/inputs/http.rb

```diff
@@ -236,19 +241,37 @@ def validate_ssl_settings!
     end
 
     if @ssl && (original_params.key?("verify_mode") && original_params.key?("ssl_verify_mode"))
-        raise LogStash::ConfigurationError, "Both 'ssl_verify_mode' and 'verify_mode' were set. Use only 'ssl_verify_mode'."
+      raise LogStash::ConfigurationError, "Both `ssl_verify_mode` and (deprecated) `verify_mode` were set. Use only `ssl_verify_mode`."
     elsif original_params.key?("verify_mode")
       @ssl_verify_mode_final = @verify_mode
-    elsif original_params.key?("ssl_verify_mode")
-      @ssl_verify_mode_final = @ssl_verify_mode
     else
       @ssl_verify_mode_final = @ssl_verify_mode
     end
 
+    if @ssl && (original_params.key?('cipher_suites') && original_params.key?('ssl_cipher_suites'))
+      raise LogStash::ConfigurationError, "Both `ssl_cipher_suites` and (deprecated) `cipher_suites` were set. Use only `ssl_cipher_suites`."
+    elsif original_params.key?('cipher_suites')
+      @ssl_cipher_suites_final = @cipher_suites
+    else
+      @ssl_cipher_suites_final = @ssl_cipher_suites
+    end
```

Comment:
I think that if `@ssl` is `false` then `@ssl_cipher_suites_final` end in:
- empty list if the user configured `cipher_suites`.
-  `SslSimpleBuilder.getDefaultCiphers` if the user in any other case. 
I think that if `ssl => false` it shouldn't set anything related to the ciphers

 I know that the method `build_ssl_params` has a guard:
```ruby
return nil unless @ssl
```
Maybe isolating the fragment with an `if @ssl`should help.

```suggestion
    if @ssl
      if original_params.key?('cipher_suites') && original_params.key?('ssl_cipher_suites')
        raise LogStash::ConfigurationError, "Both `ssl_cipher_suites` and (deprecated) `cipher_suites` were set. Use only `ssl_cipher_suites`."

      @ssl_cipher_suites_final = original_params.key?('cipher_suites') ? 
            @cipher_suites : 
            @ssl_cipher_suites
    end
```

Reply:
`ssl_cipher_suites_final` won't be used since `build_ssl_params` isn't called.
the reason why I have it this way is due existing convention above: 
```ruby
    # ...
    elsif original_params.key?("verify_mode")
      @ssl_verify_mode_final = @verify_mode
    else
      @ssl_verify_mode_final = @ssl_verify_mode
```